DEVX-2488: TIMESTAMP datatype, timezone conversion, JDBC sink connector KT (#760)
confluentinc/kafka-tutorials: _includes/tutorials/connect-sink-timestamp/ksql/markup/dev/create-sink-connector.adoc

> Now deploy a JDBC sink connector with the code below. Notice no transforms will be necessary to the `eventTime` column because the column is already a supported `TIMESTAMP` type.

Reviewer: For your consideration: maybe mention that the sink connector writes to the database.  I know most readers would get this, but a brand new user could miss something.

Author: Good point. I added a comment to address this in my latest commit. https://github.com/confluentinc/kafka-tutorials/pull/760/commits/9904d2ac9ffb1d0894c431caf88c786ba19ad091

+++++
<pre class="snippet"><code class="sql">{% include_raw tutorials/connect-sink-timestamp/ksql/code/tutorial-steps/dev/create-sink-connector.sql %}</code></pre>
+++++

[NOTE]
====

> Prior to version 0.16.0, ksqlDB did not have a `TIMESTAMP` data type so the only way to convert `BIGINT` to a `TIMESTAMP` was with Kafka Connect's Single Message Transforms (SMT), specifically the  https://docs.confluent.io/platform/current/connect/transforms/timestampconverter.html[TimestampConverter]; .

Reviewer 2: @awalther28 should this be:

```suggestion
Prior to version 0.17.0, ksqlDB did not have a `TIMESTAMP` data type so the only way to convert `BIGINT` to a `TIMESTAMP` was with Kafka Connect's Single Message Transforms (SMT), specifically the  https://docs.confluent.io/platform/current/connect/transforms/timestampconverter.html[TimestampConverter]; .
```

Now deploy a JDBC sink connector with the code below. Notice no transforms will be necessary to the `eventTime` column because the column is already a supported `TIMESTAMP` type.

+++++
<pre class="snippet"><code class="sql">{% include_raw tutorials/connect-sink-timestamp/ksql/code/tutorial-steps/dev/create-sink-connector.sql %}</code></pre>
+++++

[NOTE]
====
For ksqlDB prior to 0.16.0, the only way to convert `BIGINT` to a `TIMESTAMP` was with https://docs.confluent.io/platform/current/connect/transforms/overview.html[Single Message Transforms (SMTs)]; ksqlDB did not have a `TIMESTAMP` data type.

> If you were going this route, you would add something similar to the following to the connector config instead of performing the steps above where we convert the `EVENTTIME` column data type and convert the column to a new timezone.

Reviewer 2: ```suggestion
If you were going this route of using `TimestampConverter`, you would add the SMT into the connector configuration, something similar to the following:
```

```
"transforms": "TimestampConverter",
"transforms.TimestampConverter.type": "org.apache.kafka.connect.transforms.TimestampConverter$Value",
"transforms.TimestampConverter.format": "yyyy-MM-ddTHH:mm:ss.SSS"
"transforms.TimestampConverter.target.type": "Timestamp"
```

> Using a SMTs is simple but it has its drawbacks. First, the `TimestampConverter` SMT does not provide users a way to convert timestamp data to other timezones.

Reviewer 2: @awalther28 I brought up the "drawbacks" into the paragraph above to keep the text together, and to shorten it.  If you agree with this change, please delete these 3 lines here, and of course wordsmith the text above as you see fit.

Author: :ack: I removed those 3 lines in my latest commit and verified the KT site builds locally.

> Now create a JDBC sink connector with the code below. Notice no transforms will be necessary to the timestamp field because it is already a supported `TIMESTAMP` type.

Reviewer 2: "it is" -- what is it?

Author: addressed in the latest commit https://github.com/confluentinc/kafka-tutorials/pull/760/commits/fe37cdbb1eb4538e6474ccb6e0f8f38e283efa77

> Now create a JDBC sink connector with the code below. Notice no transforms will be necessary to the `eventTime` column because the column is already a supported `TIMESTAMP` type.

Reviewer 2: WDYT about using the word `deploy` instead of `create`?  Goal being to differentiate jar installation as done earlier with `confluent-hub install`.

Author: changed

+++++
<pre class="snippet"><code class="sql">{% include_raw tutorials/connect-sink-timestamp/ksql/code/tutorial-steps/dev/create-sink-connector.sql %}</code></pre>
+++++

> Prior to ksqlDB 0.16.0, the only way to convert a `BIGINT` or `VARCHAR` to a `TIMESTAMP` was with https://docs.confluent.io/platform/current/connect/transforms/overview.html[Single Message Transforms (SMTs)].

Reviewer 2: The information in the sentence is awesome, but the sentence feels misplaced/orphaned.  Any way to tie it in a bit more?  Either move it to another section, or keep it here and add a sentence (like...not verbatim...if you're going the SMT route, you'd add the SMT in this connector config above)

Author: hmm, thinking about where this should live.... I think near the top as part would make the most sense. Will comment once moved

Author: ![Screen Shot 2021-03-17 at 8 44 02 AM (2)](https://user-images.githubusercontent.com/12937975/111486255-f37f0300-86fc-11eb-84a2-afe63d1d5223.png)

@ybyzek what are your thoughts on above?

Reviewer 2: @awalther28 the callout is super cool 🥇   There is another drawback to SMTs -- they are not supported in Confluent Cloud -- but the KT should not draw attention to that ;) I do have minor edit suggestions, will drop it in the PR momentarily
